documentation/specs/proposed/BuildCheck-net10-polishing.md

```diff
@@ -0,0 +1,68 @@
+# BuildCheck Net10 polishing 
+
+In net 9.0 we delivered initial preview of BuildCheck feature. 
+ * Spec: https://github.com/dotnet/msbuild/blob/main/documentation/specs/BuildCheck/BuildCheck.md
+ * Work track:  https://github.com/orgs/dotnet/projects/373
+
+In addition to that we have item tracking possible improvemnts and extensions of the feature: https://github.com/dotnet/msbuild/issues/10548
+
+This doc focuses on minimal investment path that would still allow driving some adoption of BuildChecks feature and collecting feedback from real life usage.
+
+## Goals and Motivation 
+
+* Making sure the feature can be used in the real life scenarios.
+* Driving adoption.
+
+## Impact 
+
+* MSBuild team has a venue to evangelise best practices for the build.
+* Customers have a tool to formalize and distribute their view of best practices.
+* MSBuild team can be improving builds perf and security via providing new Checks notyfying about problems.
+
+
+## Stakeholders 
+- PM (@baronfel) - as a customer advocate
+- Selected internal partner repo owners (details https://github.com/dotnet/msbuild/issues/10726)
+
+### Successful handover
+- Internal partner teams ideas around BuildChecks collected and considered.
+- Selected internal partner teams are using BuildChecks in their build pipelines.
+- BuildChecks being evangelized externaly, adoption numbers grow up.
+
+## Risks 
+- Performance degradation is unacceptable on real scale project.
+- There are multiple hard to investigate or/and hard to fix bugs identified during inital adoptions.
+- Unsatisfactory specificity or/and sensitivity - leading to diminishing the valua added and endangering the adoption motivation.
+- Low perceived value of best practices enforced by the checks.
+- Confusing diagnostics/documentation leading to lowering the educational value.
+
+
+## Scope
+
+### Goals
+* Running buildchecks on partner repos, identifying and fixing issues
+* Enabling buildchecks on partner repos, providing initial support
+
+### Non-Goals
+
+* Measuring and defining the perf impact, detecting the sources of it
```

Comment:
Since perf was called out as an explicit risk, why is this a non-goal?

Reply:
Yes - perf is super important. If there is an appetite to invest any efforts into buildchecks - it should be on top of the list. If there is near to zero investment - then we need to at least keep it alive by dogfooding. Dogfooding can then fuel and justify investment into the perf.

Reply:
PerfStar also has a plan to implement measures for the BuildCheck feature, so we will be keeping an eye on it at least.